docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/ImageAnalytics/ConvertToImage.cs

```diff
@@ -68,9 +68,6 @@ private static void PrintColumns(IDataView transformedData)
                     Console.WriteLine("{0, -25} {1, -25} {2, -25}", string.Join(",", features.DenseValues().Take(5)) + "...",
                         imageObject.PhysicalDimension, string.Join(",", pixels.DenseValues().Take(5)) + "...");
                 }
-
-                // Dispose the image.
-                imageObject.Dispose();
```

Comment:
So when are those objects disposed off then? Someone somewhere has to do this... Aren't they holding handle and other windows objects?

Reply:
Can you please clarify what handle and other window objects you are referring to?

I had added that dispose call because in the past bitmap object was created such that it was holding lock on the file but that is no longer the case after this change. GC will collect the bitmap object and dispose it since there is no reference to it after sample execution. I have tested this by calling GC.Finalize(). This is sample code that should be simple and also educational. If you feel from educational point of view it makes sense to add back then I’ll do it.

Reply:
According to doc it holds the GDI+ bitmap, which is the unmanaged resource.  Since it's disposable, we should call dispose of it once it's used, because the GC might significantly underestimate the cost of the object by not tracking unmanaged resources and result in a low resource condition..  What's wrong with disposing it here?

---
In reply to: [279999356](https://github.com/dotnet/machinelearning/pull/3625#discussion_r279999356) [](ancestors = 279999356)